projects/dotcms-field-elements/src/components/dot-textfield/dot-texfield.e2e.ts

```diff
@@ -6,135 +6,282 @@ describe('dot-textfield', () => {
+                expect(await input.getProperty('disabled')).toBe(true);
+            });
+
+            it('should not render disabled attribute', async () => {
+                element.setProperty('disabled', 'false');
+                await page.waitForChanges();
+                expect(await input.getProperty('disabled')).toBe(false);
+            });
 
-            expect(spyStatusChangeEvent).toHaveReceivedEventDetail({
-                name: 'fullName',
-                status: {
-                    dotPristine: true,
-                    dotTouched: true,
-                    dotValid: true
-                }
+            it('should render disabled attribute with invalid value', async () => {
+                element.setProperty('disabled', { test: 'test' });
+                await page.waitForChanges();
+                expect(await input.getProperty('disabled')).toBe(true);
             });
         });
 
-        it('should send status and value change', async () => {
-            await input.press('a');
-            await page.waitForChanges();
-            expect(spyStatusChangeEvent).toHaveReceivedEventDetail({
-                name: 'fullName',
-                status: {
-                    dotPristine: false,
-                    dotTouched: true,
-                    dotValid: true
-                }
-            });
-            expect(spyValueChange).toHaveReceivedEventDetail({
-                name: 'fullName',
-                value: 'Johna'
+        describe('type', () => {
+            it('should set value to text on default correctly', async () => {
+                await page.waitForChanges();
+                expect(await input.getProperty('type')).toBe('text');
+            });
+
+            it('should set value correctly', async () => {
+                element.setProperty('type', 'email');
+                await page.waitForChanges();
+                expect(await input.getProperty('type')).toBe('email');
+            });
+
+            it('should render and not break when is a unexpected value', async () => {
+                element.setProperty('type', { test: true });
+                await page.waitForChanges();
+                expect(await input.getProperty('type')).toBe('text');
             });
         });
 
-        it('should emit status and value on Reset', async () => {
-            await element.callMethod('reset');
-            expect(spyStatusChangeEvent).toHaveReceivedEventDetail({
-                name: 'fullName',
-                status: {
-                    dotPristine: true,
-                    dotTouched: false,
-                    dotValid: false
-                }
-            });
-            expect(spyValueChange).toHaveReceivedEventDetail({ name: 'fullName', value: '' });
+        describe('@Events', () => {
+            beforeEach(async () => {
+                page = await newE2EPage();
+                await page.setContent(`
+                <dot-textfield
```

Comment:
Here we're testing the events with all valid props, what happen if we don't have for example name?